Add calliope recipe

sjpfenninger opened the pull request.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/calliope```) and found it was in an excellent condition.

synapticarbors commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+    - click >=3.3
+    - glpk ==4.59
+    - numexpr >=2.3.1
+    - numpy >=1.10.1
```
There's a discrepancy with the version in `setup.py`. There it says `>= 1.11`.

sjpfenninger replied:
Fixed.

synapticarbors commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+    - {{ name }} = {{ name }}.cli:cli
+
+requirements:
+  build:
```
I believe that the `--single-version-externally-managed` flag in the build script makes it such that the `install_requires` in the `setup.py` is ignored. Therefore, please limit the packages listed in this section to those that are actually required to build the recipe.

sjpfenninger replied:
Fixed.

synapticarbors commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+    - python >=3.5
+    - pyyaml >=3.11
+    - tables >=3.2.0
+  run:
```
There are some packages that are listed in `install_requires` in the `setup.py` that are missing here: `xarray` and `netcdf4`

There are also libraries that are in the `requirements.yml` that aren't included here, like `shapely`, `descartes`, `geos`, `scipy`.

Please check to ensure you have the complete list of requirements.

sjpfenninger replied:
Fixed.

synapticarbors commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+    - pyomo >=4.2,<4.3
+    - python >=3.5
+    - pyyaml >=3.11
+    - tables >=3.2.0
```
Also maybe missing `networkx`

synapticarbors commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+
+test:
+  imports:
+    - calliope
```
Perhaps also import:

```
- calliope.constraints
- calliope.lib
```

sjpfenninger commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+    - numexpr >=2.3.1
+    - numpy >=1.10.1
+    - pandas >=0.17.1,<0.18
+    - pyomo >=4.2,<4.3
```
Recent versions of ``pyomo`` are available (https://github.com/conda-forge/pyomo-feedstock/) but my requirement was for an older version.

Failing to get that older version onto conda-forge, I'm updating my recipe PR with an as yet unreleased (on PyPI) version of my package which works with the most recent pyomo version. 

I'm versioning this ``0.4.0_rc1`` as I'd like to make sure the conda-forge package is up and running for the actual and soon-to-be-released 0.4.0, since the documentation will recommend using conda and conda-forge for installation...

sjpfenninger commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,68 @@
+    - pyomo >=4.2,<4.3
+    - python >=3.5
+    - pyyaml >=3.11
+    - tables >=3.2.0
```
Fixed.

sjpfenninger commented:
@synapticarbors I've addressed your comments and it seems to build, can you check if this is now ok?

Regarding the missing ``pyomo`` feedstock, I'm repeating the response to your review comment above:

Recent versions of pyomo are available (https://github.com/conda-forge/pyomo-feedstock/) but my requirement was for an older version.

Failing to get that older version onto conda-forge, I'm updating my recipe PR with an as yet unreleased (on PyPI) version of my package which works with the most recent pyomo version.

I'm versioning this 0.4.0_rc1 as I'd like to make sure the conda-forge package is up and running for the actual and soon-to-be-released 0.4.0, since the documentation will recommend using conda and conda-forge for installation...

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/calliope```) and found it was in an excellent condition.

jakirkham commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,71 @@
+
+about:
+  home: https://www.callio.pe/
+  license: Apache-2.0
```
Could add `license_family: Apache`.

sjpfenninger replied:
done

jakirkham commented on recipes/calliope/meta.yaml:
```diff
@@ -0,0 +1,71 @@
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip: True  # [py<35]
+  entry_points:
+    - {{ name }} = {{ name }}.cli:cli
```
Templating here should probably be avoided.

sjpfenninger replied:
done

jakirkham commented:
> Recent versions of pyomo are available (https://github.com/conda-forge/pyomo-feedstock/) but my requirement was for an older version. This is now fixed.

Meaning this is no longer necessary?

sjpfenninger commented:
@jakirkham:

> Meaning this is no longer necessary?

The dependency is still necessary, but the recipe now uses a more recent version already available on conda-forge, so no further additions to conda-forge are necessary.

jakirkham commented:
Thanks @sjpfenninger .
